### ISSUE TITLE
Katib v0.15.0 Roadmap

### ISSUE THREAD

**Issue by johnugeorge** [kubeflow/katib#1993]
Major features tracked for next Katib release. Few might be a stretch but still tracked for next release. 

**Kubeflow Pipelines**

 - [ ] Better data passing for Katib component - ([#1846](https://github.com/kubeflow/katib/issues/1846))
 - [ ] Use Kubeflow Pipelines as Trial target - ([#1914](https://github.com/kubeflow/katib/issues/1914))

**MLFlow Integration**
 - [ ] Support Push based metric collection([#577](https://github.com/kubeflow/katib/issues/577))

**UI enhancements**
 - [ ] Metrics out of view for experiment metrics (#1923)
 - [x] 500 error code after refresh https://github.com/kubeflow/katib/issues/1967
 - [ ] Expose trial logs(#971 )
 - [x] Sorting trial list(#1441 )
 - [ ] Expose more status fields ( #1434 )

**Misc**
 - [ ] Narrow RBAC rules for Katib deployment([#1639](https://github.com/kubeflow/katib/issues/1639))
 - [ ] Customize Suggestion deployment([#1737](https://github.com/kubeflow/katib/issues/1737) [#1976](https://github.com/kubeflow/katib/issues/1976))
 - [x] Support for Kubernetes 1.25 (#1957)
 - [ ] Support Log uniform scale (#1207 )
 - [ ] Implement Conformance tests  (#2044)

/cc @tenzen-y @andreyvelich @gaocegege

**Comment by johnugeorge** [comment]
@tenzen-y @anencore94 @andreyvelich Please add to the list if I missed out any.

**Comment by tenzen-y** [comment]
@johnugeorge Thanks for creating this!
I would like to add #1207 to the next katib release. WDYT?

**Comment by tenzen-y** [comment]
Also, I think that we need to add postponed https://github.com/kubeflow/katib/pull/1728#issuecomment-959328401.

**Comment by johnugeorge** [comment]
> @johnugeorge Thanks for creating this!
> I would like to add #1207 to the next katib release. WDYT?

Added.

**Comment by anencore94** [comment]
we aim to this features are supposed to be implemented before `Wednesday, Jan 25th 2023: Week 18 - Feature Freeze` right ?

**Comment by andreyvelich** [comment]
I have the following thoughts:

- We need to promote `tune` API: https://github.com/kubeflow/katib/pull/1951 in the new release to simplify Katib usage and attract more users.
- @johnugeorge @kimwnasptd Are we targeting the above UI changes before **Jan 25th** ?
- @tenzen-y @johnugeorge I want to have broader discussion around `log-uniform` and other HPs distributions before starting the API changes: https://github.com/kubeflow/katib/issues/1207. It would be great if we could discuss it on the upcoming AutoML meeting (**Nov 16th**).

**Comment by andreyvelich** [comment]
@tenzen-y Sure, we are going to share the updates.

> Also, I think that we might be good to extend API based on https://github.com/kubeflow/katib/issues/1207#issuecomment-639275103.

If we are going to add distribution to FeasibleSpace, maybe we should deprecate [ParameterType in our APIs](https://github.com/kubeflow/katib/blob/master/pkg/apis/controller/experiments/v1beta1/experiment_types.go#L194), since it's redundant.

e.g. Hyperopt directly set `uniform`, `loguniform`, etc. parameter with min and max value: http://hyperopt.github.io/hyperopt/getting-started/search_spaces/

**Comment by tenzen-y** [comment]
> If we are going to add distribution to FeasibleSpace, maybe we should deprecate [ParameterType in our APIs](https://github.com/kubeflow/katib/blob/master/pkg/apis/controller/experiments/v1beta1/experiment_types.go#L194), since it's redundant.

sgtm 
This is just a question, in that case, do we handle with `categorical` and `discrete` as distribution?

**Comment by andreyvelich** [comment]
> This is just a question, in that case, do we handle with categorical and discrete as distribution?

I think, we don't need `discrete` at all, since for the suggestion services (at least of one that we used) `categorical` and `discrete` is the same (some frameworks call it `choice`).

**Comment by tenzen-y** [comment]
> > This is just a question, in that case, do we handle with categorical and discrete as distribution?
> 
> I think, we don't need `discrete` at all, since for the suggestion services (at least of one that we used) `categorical` and `discrete` is the same parameter (some frameworks call it `choice`).

@andreyvelich Thanks for clarifying. I agree with your idea.

**Comment by tenzen-y** [comment]
Around next month, I will propose the API design for #1207.

**Comment by tenzen-y** [comment]
> @tenzen-y If you want, you can create small proposal under docs: https://github.com/kubeflow/katib/tree/master/docs/proposals

Thanks for letting me know. Looks great!
I will submit a small enhancement proposal!

**Comment by tenzen-y** [comment]
#2055

**Comment by andreyvelich** [comment]
The Katib 0.15 has been officially released: https://github.com/kubeflow/katib/releases/tag/v0.15.0

Thank you very much for all of your help!
